scripts/0-preinstall.sh

```diff
@@ -91,48 +71,83 @@ subvolumesetup () {
 # unmount root to remount with subvolume 
     umount /mnt
 # mount @ subvolume
-    mount -o ${MOUNT_OPTIONS},subvol=@ ${partition3} /mnt
+    mount -o ${MOUNT_OPTIONS},subvol=@ ${rootpartition} /mnt
 # make directories home, .snapshots, var, tmp
     mkdir -p /mnt/{home,var,tmp,.snapshots}
 # mount subvolumes
     mountallsubvol
 }
 
-if [[ "${DISK}" =~ "nvme" ]]; then
-    partition2=${DISK}p2
-    partition3=${DISK}p3
-else
-    partition2=${DISK}2
-    partition3=${DISK}3
-fi
-
-if [[ "${FS}" == "btrfs" ]]; then
-    mkfs.vfat -F32 -n "EFIBOOT" ${partition2}
-    mkfs.btrfs -L ROOT ${partition3} -f
-    mount -t btrfs ${partition3} /mnt
-    subvolumesetup
-elif [[ "${FS}" == "ext4" ]]; then
-    mkfs.vfat -F32 -n "EFIBOOT" ${partition2}
-    mkfs.ext4 -L ROOT ${partition3}
-    mount -t ext4 ${partition3} /mnt
-elif [[ "${FS}" == "luks" ]]; then
-    mkfs.vfat -F32 -n "EFIBOOT" ${partition2}
+formatandmount () {
+    if [[ "${FS}" == "btrfs" ]]; then
+        mkfs.btrfs -L ROOT ${rootpartition} -f
+        mount -t btrfs ${rootpartition} /mnt
+        subvolumesetup
+    elif [[ "${FS}" == "ext4" ]]; then
+        mkfs.ext4 -L ROOT ${rootpartition}
+        mount -t ext4 ${rootpartition} /mnt
+    elif [[ "${FS}" == "luks" ]]; then
 # enter luks password to cryptsetup and format root partition
-    echo -n "${LUKS_PASSWORD}" | cryptsetup -y -v luksFormat ${partition3} -
+        echo -n "${LUKS_PASSWORD}" | cryptsetup -y -v luksFormat ${rootpartition} -
 # open luks container and ROOT will be place holder 
-    echo -n "${LUKS_PASSWORD}" | cryptsetup open ${partition3} ROOT -
+        echo -n "${LUKS_PASSWORD}" | cryptsetup open ${rootpartition} ROOT -
 # now format that container
-    mkfs.btrfs -L ROOT ${partition3}
+        mkfs.btrfs -L ROOT ${rootpartition}
 # create subvolumes for btrfs
-    mount -t btrfs ${partition3} /mnt
-    subvolumesetup
+        mount -t btrfs ${rootpartition} /mnt
+        subvolumesetup
 # store uuid of encrypted partition for grub
-    echo ENCRYPTED_PARTITION_UUID=$(blkid -s UUID -o value ${partition3}) >> $CONFIGS_DIR/setup.conf
+        echo ENCRYPTED_PARTITION_UUID=$(blkid -s UUID -o value ${rootpartition}) >> $CONFIGS_DIR/setup.conf
+    fi
+}
+
+umount -A --recursive /mnt # make sure everything is unmounted before we start
+
+if [[ $INSTALL_IN = "PART" ]]; then # Checking if install to partition
+    EFIpartition=${BOOTPART}
+    rootpartition=${PART}
+
+    if [[ -d "/sys/firmware/efi" ]]; then # Checking for UEFI system
+        if [[ $FORMATEFI = "yes" ]]; then
+            mkfs.vfat -F32 -n "EFIBOOT" ${EFIpartition}
+        fi
+        formatandmount
+        # mount EFI partition
+        mkdir -p /mnt/boot/EFI
+        mount -t vfat ${EFIpartition} /mnt/boot/EFI
+    elif [[ $(sudo sudo fdisk -l | grep -i '^Disklabel type') = "Disklabel type: gpt" ]]; then # Checking for GPT Disk Label on a Legacy BIOS (non UEFI) System
```

Comment:
hey. I think you have to modify this as well.

Reply:
I totally forgot about that. Its done.